src/praisonai/praisonai/cli_backends/__init__.py

```
@@ -6,6 +6,44 @@
+        
+    Returns:
+        CliBackendProtocol instance or None
+        
+    Raises:
+        ValueError: If configuration is invalid
+    """
+    if not value:
+        return None
+        
+    # If already an instance, return as-is (duck typing)
+    if hasattr(value, '__call__') or hasattr(value, 'process_turn'):
```

Ruff (0.15.15)

[warning] Using `hasattr(x, "__call__")` to test if x is callable is unreliable. Use `callable(x)` for consistent results.

Replace with `callable()`

(B004)